multimodal/src/autogluon/multimodal/utils/environment.py

```diff
@@ -347,3 +351,15 @@ def _get_mmlab_installation_guide(package_name):
         raise ValueError("Available package_name are: mmdet, mmcv, mmengine.")
 
     return err_msg
+
+
+def run_ddp_only_once(num_gpus, strategy):
+    if strategy in DDP_STRATEGIES:
+        global FIRST_DDP_RUN  # Use the global variable to make sure it is tracked per process
```

Comment:
Is this global variable shared across processes? If ddp creates 1 additional process running from the beginning, the new process would be set `num_gpus` 1 and strategy `auto` when running this function?

Reply:
The new DDP process can't see the previous global variable.